numpy/core/src/multiarray/descriptor.c

```diff
@@ -1115,14 +1111,8 @@ _convert_from_dict(PyObject *obj, int align)
             Py_DECREF(ind);
             goto fail;
         }
-        if (align) {
-            ret = PyArray_DescrAlignConverter(descr, &newdescr);
-        }
-        else {
-            ret = PyArray_DescrConverter(descr, &newdescr);
-        }
-        Py_DECREF(descr);
```

Comment:
This DECREF seems to have gone lost? Won't be the reason for the test failure though, puzzles me as well right now.

Reply:
Good catch, but the travis failure persists after the fix.